lib/getHashDigest.js

```diff
@@ -36,15 +43,35 @@ function encodeBufferToBase(buffer, base) {
 	return output;
 }
 
+function getBase(digestType) {
+	return digestType === BASE_EMOJI ? emojiList.length : digestType.substr(4);
+}
+
+function sliceEmojis(output, maxLength) {
+	let match = emojiRegex.exec(output);
+	const emojis = [];
+	while(match) {
+		emojis.push(match[0]);
+		match = emojiRegex.exec(output);
+	}
+	return emojis.splice(0, maxLength).join("");
+}
+
+function sliceOutput(output, digestType, maxLength) {
+	if(digestType === BASE_EMOJI) {
+		return sliceEmojis(output, maxLength);
+	}
+	return output.substr(0, maxLength);
+}
+
 function getHashDigest(buffer, hashType, digestType, maxLength) {
 	hashType = hashType || "md5";
 	maxLength = maxLength || 9999;
 	const hash = require("crypto").createHash(hashType);
 	hash.update(buffer);
-	if(digestType === "base26" || digestType === "base32" || digestType === "base36" ||
-		digestType === "base49" || digestType === "base52" || digestType === "base58" ||
-		digestType === "base62" || digestType === "base64") {
-		return encodeBufferToBase(hash.digest(), digestType.substr(4)).substr(0, maxLength);
+	if(digestType && !!baseEncodeTables[getBase(digestType)]) {
```

Comment:
Maybe add a const array of valid `digestType`s?

Reply:
This was the idea with using`baseEncodeTables` as a const to not have to repeat the digestTypes. Lets see 😄